apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/consumer/MultipleChannelsConsumer.java

```diff
@@ -30,13 +32,12 @@
 public class MultipleChannelsConsumer extends Thread {
     private volatile boolean running;
     private volatile ArrayList<Group> consumeTargets;
-    @SuppressWarnings("NonAtomicVolatileUpdate")
-    private volatile long size;
+    private final AtomicLong size = new AtomicLong();
```

Comment:
Why need atomic long? We don't do concurrency consuming in any case

Reply:
Yes, there has another way to fix this warning. Just add `synchronized` on the `MultipleChannelsConsumer#addNewTarget`, but it has already added on `BulkConsumePool#add` which invoke `MultipleChannelsConsumer#addNewTarget`.

IDE shows warning because the volatile filed has non-atomic operations on this Class, the non-atomic operations should be synchronized. Or we just use AtomicLong.

Reply:
I don't find the real case, the reason for warning is this could happen, but in our case, there isn't.

Reply:
It doesn't mean there has a wrong logic in the release code, maybe it runs well. But I just want to indicate that was a bad practice to use `volatile` in this way. 

If there are multi-threads invoke the `MultipleChannelsConsumer#addNewTarget` method and the field `size` still uses `volatile`,  the `size` maybe a wrong value. Because 
> The Java `volatile` keyword guarantees visibility only! 

If you don't want to use AtomicLong, I am willing to just add `synchronized` on the `MultipleChannelsConsumer#addNewTarget` method.

Reply:
We are trying to avoid any possible lock, it is also slower, more or less. That is why it may be not a good idea.
SkyWalking codes are for product environment and high performance, we have no plan to use the codes to guide people how to write codes.

So, the key is, if the current version of codes has a bug, we are open to the solution.

Reply:
Okay, I can revert `MultipleChannelsConsumer` first. 
But the `SimpleRollingPartitioner` has a bug, and I have added a test case. Please take a look.